src/dumpling-service/dumpling.dataworker/DumplingDataWorkerRole.cs

```diff
@@ -0,0 +1,107 @@
+﻿using System.Diagnostics;
+using System.Net;
+using System.Threading;
+using Microsoft.WindowsAzure.ServiceRuntime;
+
+
+namespace dumplingDataWorker
+{
+    using DumplingLib;
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Text;
+    using System.Threading.Tasks;
+    public class DumplingDataWorkerRole : RoleEntryPoint
+    {
+        
+        public DumplingDataWorkerRole()
+        {
+            _dataHandler = new Dictionary<string, DataHandler>()
+            {
+                { "check_if_helix_worker", DataHandlers.CheckIfHelixId },
+                { "sql_index_target", DataHandlers.SqlIndexTarget }
+            };
+        }
+
+        public override void Run()
+        {
+            DumplingEventHub.FireEvent(new DataWorkerRunEvent()).Wait();
+            Trace.WriteLine("Starting processing of messages");
+
+            
+            // start a background task that queries Helix for data.
+             Task.Run(HelixEventProxy.PollTrackedHelixWorkItems); 
+
+            // initiates the message pump and callback is invoked for each message that is received, calling close on the client will stop the pump.
+            DumplingDataWorkerQueueController.Queue.OnMessageAsync(async (receivedMessage) =>
+                {
+                    Trace.WriteLine("processing service bus message: " + receivedMessage.SequenceNumber.ToString());
+                    try
+                    {
+                        // Process the message
+                        if (receivedMessage.Properties.ContainsKey("data_type") && (receivedMessage.Properties["data_type"] is string)) 
+                        {
+                            var key = receivedMessage.Properties["data_type"] as string;
+                            var dumplingId = receivedMessage.Properties["dumpling_id"] as string;
+                            var body = receivedMessage.GetBody<Stream>();
+
+                            await receivedMessage.CompleteAsync();
+
+                            // TODO: Debug/Understand: the received message is disposed of after the await call returns. Is this a framework/runtime bug?
```

Comment:
TODO: Remove this TODO. I learned that I had to use OnMessageAsync(...) instead of OnMessage(...) in order to avoid the runtime disposing of my message during from an await.